app/controllers/spree/api/ams/orders_controller.rb

```diff
@@ -4,6 +4,15 @@ module Ams
       class OrdersController < Spree::Api::OrdersController
         include Serializable
         include Requestable
+
+        def mine
+          if current_api_user.persisted?
+            @orders = current_api_user.orders.reverse_chronological
+            respond_with(@orders)
+          else
+            render "spree/api/errors/unauthorized", status: :unauthorized
+          end
+        end
```

Comment:
Ransack may also be able to handle the loading of specific IDs into a collection for you